src/lib.rs

```diff
@@ -158,21 +158,177 @@ mod core {
 use core::fmt;
 
 #[cfg(not(feature = "std"))]
+#[macro_use]
 extern crate alloc;
+
 #[cfg(not(feature = "std"))]
 use alloc::Vec;
+#[cfg(not(feature = "std"))]
+use alloc::string::String;
+#[cfg(not(feature = "std"))]
+use alloc::borrow::ToOwned;
+#[macro_use]
+extern crate serde_derive;
+
+extern crate serde;
+
+use serde::ser::{SerializeStruct, SerializeSeq, Serialize, Serializer};
+
+/// Types allowed to be value in the context vector
+#[derive(Debug, Clone)]
+pub enum ThrowContextValues {
+    ///Boolean
+    Bool(bool),
+    ///Int8
+    Int8(i8),
+    ///Uint8
+    Uint8(u8),
+    ///Int16
+    Int16(i16),
+    ///Uint16
+    Uint16(u16),
+    ///Int32
+    Int32(i32),
+    ///Uint32
+    Uint32(u32),
+    ///Int64
+    Int64(i64),
+    ///Uint64
+    Uint64(u64),
+    ///Float32
+    Float32(f32),
+    ///Float64
+    Float64(f64),
+    ///String
+    String(String),
+}
+
+impl Serialize for ThrowContextValues {
```

Comment:
Does implementing serialize manually here provide much benefit over `#[derive(Serialize)]` with `#[serde(untagged)]`?

From what I can tell they should be the same serialization. If this is the case, I think we should prefer derive for less code to maintain.

Reply:
done! a lot of code was delete in the process